### PR TITLE
Fix Travis

### PR DESCRIPTION
**Pull request by sjakobi**
This fixes building the tests with GHC-7.8 by removing the constraints to build with the installed versions of `containers` and `binary`.

Benchmark builds are simplified by switching from `criterion` to `gauge`.

This also adds jobs with GHC-8.6 and 8.8.

### CODE REVIEW

File: benchmarks/unordered-containers-benchmarks.cabal [hunk 1]
```diff
@@ -23,7 +23,7 @@ executable unordered-containers-benchmarks
     base,
     bytestring,
     containers,
-    criterion,
+    gauge,
```
Comment (sjakobi) [inline benchmarks/unordered-containers-benchmarks.cabal]: Not sure whether this file can be removed now that the dependency cycle has been removed?

**Comment by sjakobi** [comment]
Ping! :)

**Comment by treeowl** [comment]
Please squash into one or a very few logical commits. Please justify the switch to `gauge` in a commit message.

**Comment by sjakobi** [comment]
> Please squash into one or a very few logical commits. Please justify the switch to `gauge` in a commit message.

Done!

**Comment by treeowl** [comment]
I don't love losing features relative to Criterion, but the dependency cycles are admittedly a bit wonky to deal with. Fine.

**Comment by sjakobi** [comment]
@treeowl Thanks for reviewing and merging!

Which `criterion` features are you referring to?

**Comment by treeowl** [comment]
Pretty graphs, mostly.

**Comment by sjakobi** [comment]
> Pretty graphs, mostly.

You could give https://github.com/composewell/bench-show a try.

**Pull request merged**